packages/ai-core/src/common/prompt-variable-contribution.ts

```diff
@@ -81,7 +81,12 @@ export class PromptVariableContribution implements AIVariableContribution, AIVar
                 }
             }
         }
-        return undefined;
+        console.debug(`Could not resolve prompt variable '${request.variable.name}' with arg '${request.arg}'. Returning empty string.`);
```

Comment:
I think it'd probably be better to use `ILogger` instead of plain console.

```ts
@inject(ILogger)
protected logger: ILogger;

...

this.logger.debug(`Could not resolve prompt variable '${request.variable.name}' with arg '${request.arg}'. Returning empty string.`);
```